src/index.ts

```diff
@@ -0,0 +1,77 @@
+const validVersions = [
+	'es3',
+	'es5',
+	'es2015',
+	'es2016',
+	'es2017',
+	'es2018',
+	'es2019',
+	'es2020',
+	'esnext',
+] as const;
+
+export type ValidVersion = typeof validVersions[number];
+
+let desiredVersionIndex = 0;
+
+export const set = (version: ValidVersion) => {
+	const versionIndex = validVersions.indexOf(version);
+	if (versionIndex === -1) {
+		throw new Error('Invalid es version to set: ' + version);
+	}
+	desiredVersionIndex = versionIndex;
+};
+
+export const get = (supportedVersions?: ValidVersion[]) => {
+	if (supportedVersions == null) {
+		// If no supported version array is passed just return the desired version
+		return validVersions[desiredVersionIndex];
+	}
+
+	// Otherwise check for the closest matching supported version and return that, priority is:
+	// 1st: exact match
+	// 2nd: closest match under
+	// 3rd: closest match over
+	// 4th: error
+	let closestVersionUnder: number | undefined;
+	let closestVersionOver: number | undefined;
+	for (const supportedVersion of supportedVersions) {
+		const supportedVersionIndex = validVersions.indexOf(supportedVersion);
+		if (supportedVersionIndex === desiredVersionIndex) {
+			// 1st priority: exact match
+			return validVersions[desiredVersionIndex];
+		}
+		if (supportedVersionIndex !== -1) {
```

Comment:
I would personally check for `=== -1` first and throw right away if that's the case, since it could just be a typo.
Moreover this will make the code shallower.

Reply:
The reason I did it like this is that if any are invalid I'd want it to always throw, which means that we'd need to first check every version

Reply:
I would still prefer the symmetry of having both methods throw on invalid options, especially since now with the enum the second loop will be both fast and clean, but I'm not going to insist on this if you still prefer to not throw for future versions.